tests/ostree-spawn.c

```diff
@@ -170,144 +181,150 @@ ostree_remote_add (GFile *repo,
+                CmdResult *cmd,
                 GError **error)
 {
-  SSDEF;
-  g_auto(GStrv) args = NULL;
-
-  args = generate_strv (SS (g_file_get_path (sysroot)),
-                        NULL);
+  g_autofree gchar *raw_sysroot_path = g_file_get_path (sysroot);
+  gchar *args[] =
+    {
+      (gchar *)raw_sysroot_path,
+      NULL
+    };
 
   return ostree_admin_spawn_in_sysroot (sysroot,
-					"init-fs",
-					args,
-					cmd,
-					error);
+                                        "init-fs",
+                                        args,
+                                        cmd,
+                                        error);
 }
 
 gboolean
 ostree_os_init (GFile *sysroot,
                 const gchar *remote_name,
-                CmdStuff *cmd,
+                CmdResult *cmd,
                 GError **error)
 {
-  g_auto(GStrv) args = NULL;
-
-  args = generate_strv (remote_name,
-                        NULL);
+  gchar *args[] =
+    {
+      (gchar *)remote_name,
+      NULL
+    };
 
   return ostree_admin_spawn_in_sysroot (sysroot,
-					"os-init",
-					args,
-					cmd,
-					error);
+                                        "os-init",
+                                        args,
+                                        cmd,
+                                        error);
 }
 
 gboolean
 ostree_status (GFile *sysroot,
-               CmdStuff *cmd,
+               CmdResult *cmd,
                GError **error)
 {
   return ostree_admin_spawn_in_sysroot (sysroot,
-					"status",
-					NULL,
-					cmd,
-					error);
+                                        "status",
+                                        NULL,
+                                        cmd,
+                                        error);
 }
 
 gboolean
 ostree_httpd (GFile *served_dir,
+              GFile *port_file,
               guint16 *port,
-              CmdStuff *cmd,
+              CmdResult *cmd,
               GError **error)
 {
-  SSDEF;
-  g_auto(GStrv) argv = NULL;
-  g_autoptr(GFileIOStream) stream = NULL;
-  g_autoptr(GFile) tmp_port = g_file_new_tmp ("eos-updater-test-httpd-port-XXXXXX", &stream, error);
-  g_autoptr(GBytes) bytes = NULL;
-  g_autofree gchar *port_contents = NULL;
-  gchar *endptr;
-  guint64 number;
-  int saved_errno;
-
-  if (tmp_port == NULL)
-    return FALSE;
-
-  argv = generate_strv (OSTREE_BINARY,
-                        "trivial-httpd",
-                        "--autoexit",
-                        "--daemonize",
-                        SS (flag ("port-file", SS (g_file_get_path (tmp_port)))),
-                        SS (g_file_get_path (served_dir)),
-                        NULL);
+  g_autofree gchar *raw_port_file = g_file_get_path (port_file);
+  g_autofree gchar *raw_served_dir = g_file_get_path (served_dir);
+  CmdArg args[] =
+    {
+      { NULL, OSTREE_BINARY },
+      { NULL, "trivial-httpd" },
+      { "autoexit", NULL },
+      { "daemonize", NULL },
+      { "port-file", raw_port_file },
```

Comment:
> > -  argv = generate_strv (OSTREE_BINARY,
> > -                        "trivial-httpd",
> > -                        "--autoexit",
> > -                        "--daemonize",
> > -                        SS (flag ("port-file", SS (g_file_get_path (tmp_port)))),
> > -                        SS (g_file_get_path (served_dir)),
> > -                        NULL);
> > +  g_autofree gchar *raw_port_file = g_file_get_path (port_file);
> > +  g_autofree gchar *raw_served_dir = g_file_get_path (served_dir);
> > +  CmdArg args[] =
> > +    {
> > +      { NULL, OSTREE_BINARY },
> > +      { NULL, "trivial-httpd" },
> > +      { "autoexit", NULL },
> > +      { "daemonize", NULL },
> > +      { "port-file", raw_port_file },
>
> Port file is written by ostree trivial-httpd with the port number it is
> listening on. We just read it later. So nothing has changed here, other
> that the port file reading is outside this function now.
>

Ah okay that makes sense. So the file is dynamically generated during test
time and it needs to be passed from one process to another (and the test
process won't immediately have access to it). That makes sense to me, happy
to keep that change.